### ISSUE TITLE
Issue trying to click a RadioButton

### ISSUE THREAD

**Issue by onassisronaldo** [HtmlUnit/htmlunit#117]
Have a good one!

Yesterday i posted this question on stackoverflow https://stackoverflow.com/questions/59670380/unable-to-click-an-radiobutton-with-htmlunit
and @rbri  asked me to make this issue post here, so here it goes. (thanks for you time)

I'm converting this already done code from Selenium using IE drive to an HTMLUnit one. The code works fine on Selenium and i'm also able to test the original application through browser. The main issue is at an specific form. 

There's this button in the page that i click to generate a List, which isn't happening, it retrieves me an error saying that i haven't clicked the previous RadioButton in the form, which i have clicked (with all the possible methods available).

Here are some code snippets:

----Part in which i try to click the button, the executeScript lines where retrieved from the previous Selenium working code i said.

```
//printlnlog(" --- Selecting client type ---);

HtmlRadioButton Input rdbCliente = (HtmlRadioButton Input) supportPage.getElementById("rdlbClient");
supportPage = rdbClient.click();

supportPage.executeJavaScript (document.getElementById('rdbClient').click;

supportPage.executeJavaScript("document.getElementById( rdbClient').setAttribute('checked',true)");
supportPage.executeJavaScript(" "
	+ " f_SelectMainContact('C');"
	+ " bclose = true; "
	+ " document.getElementById('Acao').value = "ICONT'; "
	+ " document.getElementById('retorno').value = 'OK|C'; " 
	+ " document.Form1.submit(): ");

printlnLog("Test: " + supportPage.getElementsById("rdoClient).toString());

```
-----The contect of the RadioButton i retrieved from the Browser inspection tool

`<input name="group1" id="rdbClient" onmousedown="javascript:SetMouseButton(event.button)" onclick=javascrip:TypeSelectedContact();" type="radio" value="C"></input> Client
`
----------------------------------------------------------------------------------------------------------

I strongly believe that it is an JS error, that the button is actually being clicked, though it doesn't do anything, at some tests the RadioButton even received the "checked" attribute, though its JS code haven't been executed. Also tried adding millions of method for waiting on JS to execute.

*sorry about didn't adding a lot of code, the pc which has it doesnt have internet access so i gotta type it all again here...

Thanks for your time again.

**Comment by rbri** [comment]
Lets start with some suggestions
* do not try to execute js yourself, usually the safer way is to interact with the controls like rdbClient.click();
* submit the form the same way - by clicking the appropriate button
* check the results with page.asXml()
* keep in mind the interaction methods like click() returning the sync result -if the js starts a async page update do something like

page = elem.click();
webClient.waitForBackgroundJavaScript(10000);
page = page.getEnclosingWindow().getEnclosedPage();
page.asXml()

The last step is the trick in many cases....

**Comment by onassisronaldo** [comment]
Hey there @rbri, sorry for the long period without replying, its been a harsh time around the code, without any progress...

I've tried your suggestions, but none seemed to work. i tried waiting on the JavaScript, requested it through the Buttons instead of manually, submited the form, checked the page asXml, though it just won't generate the table. 

The pages i'm trying to work with are in .aspx (considerably old) and are "onbyfly". As far as i know, they don't actually alter the page itself, they are dynamic, the updates are only in the browser's cache during the session, does HtmlUnit support that kind of functions, if so, how can i do it? 


Anyway, thanks again for the help, i appreciate it!

**Comment by rbri** [comment]
I there any chance for me to access the application?

**Comment by rbri** [comment]
Without any option to have a look (and debug) with the real application i see no chance to fix this.
Any option to provide some kind of sample code to reproduce the problem?

**Comment by rbri** [comment]
no feedback - will close this; please open a new one if you like to work on this again
